src/main/java/edu/illinois/cs/dt/tools/diagnosis/pollution/Pollution.java

```diff
@@ -62,7 +62,10 @@ protected void save() {
 
             StaticFieldPathManager.createModePath(TracerMode.FIRST_ACCESS);
 
-            if (minimized.deps().isEmpty()) {
+            if (minimized.polluters().isEmpty()) {
+                return new HashMap<>();
+            }
+            if (minimized.polluters().get(0).deps().isEmpty()) {
```

Comment:
Does this ever happen, that there's is an entry in the polluters but it's empty? In either case, I feel this should probably be extracted to a method inside the MinimizedTestResult.